R/model-function.R

```diff
@@ -84,10 +88,21 @@ monty_model_function <- function(density, packer = NULL, fixed = NULL,
   properties <- monty_model_properties(
     allow_multiple_parameters = allow_multiple_parameters)
 
+  parameters <- packer$names()
+
+  use_domain <- !is.null(domain)
+  if (use_domain) {
+    domain <- validate_domain(domain, parameters, call = call)
+  }
+
   monty_model(
-    list(parameters = packer$names(),
+    list(parameters = parameters,
          density = function(x) {
+           if (use_domain && any(x < domain[, 1] | x > domain[, 2])) {
```

Comment:
This needs more care if `allow_multiple_parameters` is `TRUE` and `x` is a matrix